Keas.Mvc/ClientApp/components/Access/AssignmentContainer.tsx

```diff
@@ -0,0 +1,263 @@
+import RevokeAccess from './RevokeAssigment';
+import { Context } from '../../Context';
+import AccessList from './AccessList';
+import AssignAccess from './AssignAccess';
+
+// List of assignments passed by props, since this container can be in multiple places
+interface IProps extends RouteChildrenProps<IMatchParams> {
+  onRevokeSuccess?(assignment: IAccessAssignment);
+  onAssignSuccess?(access: IAccessAssignment);
+  disableEditing?: boolean;
+  person?: IPerson;
+  access?: IAccess;
+}
+
+interface IState {
+  isRevokeModalShown: boolean;
+  isAssignModalShown: boolean;
+  selectedAssignment?: IAccessAssignment;
+  assignments: Array<IAccessAssignment>;
+}
+
+class AssignmentContainer extends React.Component<IProps, IState> {
+  public static contextType = Context;
+  public context!: React.ContextType<typeof Context>;
+
+  constructor(props: IProps) {
+    super(props);
+
+    const assignments =
+      (props.access &&
+        props.access.assignments.map(assignment => ({
+          ...assignment,
+          access: props.access
+        }))) ||
+      [];
+    const { action } = props.match.params;
+
+    this.state = {
+      isRevokeModalShown: !props.disableEditing && action === 'revoke',
+      isAssignModalShown: !props.disableEditing && action === 'assign',
+      assignments: assignments,
+      selectedAssignment: assignments.find(
+        el => el.id === parseInt(props.match.params.id, 10)
+      )
+    };
+  }
+
+  // assume that props.person is valid
+  async fetchAssignments(): Promise<IAccessAssignment[]> {
+    const accessFetchUrl = `/api/${this.context.team.slug}/access/listAssigned?personId=${this.props.person.id}`;
+    let accesses: IAccess[] = null;
+    try {
+      accesses = await this.context.fetch(accessFetchUrl);
+    } catch (err) {
+      toast.error('Error loading access list. Please refresh and try again.');
+    }
+
+    return accesses.map(access => ({
+      ...access.assignments.find(
+        assignment => assignment.accessId === access.id && assignment.personId === this.props.person.id
+      ),
+      access
+    }));
+  }
+
+  async componentDidMount() {
+    if (this.state.assignments.length === 0) {
+      let assignments = await this.fetchAssignments();
+      console.log(assignments)
+      this.setState({
+        assignments: assignments,
+        selectedAssignment: assignments.find(
+          el => el.id === parseInt(this.props.match.params.id, 10)
+        )
+      });
+    }
+  }
+
+  private showRevokeModal = (assignment: IAccessAssignment) => {
+    this.setState({
+      isRevokeModalShown: true,
+      selectedAssignment: assignment
+    });
+    this.props.history.push(
+      `${this._getBaseUrl()}/accessAssignment/revoke/${assignment.id}`
+    );
+  };
+
+  private hideModals = () => {
+    this.setState({
+      isRevokeModalShown: false,
+      isAssignModalShown: false,
+      selectedAssignment: null
+    });
+    this.props.history.replace(this._getBaseUrl());
```

Comment:
is there a reason you're doing replace instead of push here?

Reply:
My reasoning was that they shouldn't be able to use the back button to bring up the modal again, since it has already been revoked.